Event

whwang99 opened the pull request.

Commit:
🚶🐀 Checkpoint
./index.html:22217575/607

Commit:
🐸🎓 Checkpoint
./index.html:22217575/350

Commit:
🍂🏵 Checkpoint
./index.html:22217575/1

Commit:
🐯👠 Checkpoint
./index.html:22217575/160

Commit:
🕰🏂 Checkpoint
./index.html:22217575/5926

Commit:
⏳🍗 Checkpoint
./index.html:22217575/36195

Commit:
🌃☃️ Checkpoint
./index.html:22217575/41490

Commit:
loaded information into html

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,168 @@
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="styles.css" />
+    <title>project3</title>
```
Consider using the conference name as the `title` element?

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,168 @@
+      Mind by SEED magazine.
+    </p>
+  </body>
+</html>
```
Your current markup is looking good! See my additional comment so you know which elements are currently missing 👍🏻

angeliquejw replied:
Markup continued to be clear and logically organized!

angeliquejw replied:
Also, FYI -- because I can still 👀 the markup as you're currently working on it, you're currently doing something like this in your HTML for your talks list:

```html
<ul>
    <div class="talks-card-display">
        <li class="talks-card">
            ...lots of content here
        </li>
    </div>
</ul>
```
Note that this is invalid HTML -- the only thing that can be _directly_ inside an `ul` or `ol` is a `li` element. Inside the `li` element you can have all sorts of markup, including `div` and `img`, etc.  So, you may want to swap things around so the `talks-card-display` is your `li` and there's a `div` inside it instead.

(Again, this isn't the markup I'm currently seeing on GitHub and doesn't affect your grade. It was only looking at your current work on Glitch that I spotted this issue.)

angeliquejw reviewed (commented):
Reminder of the project requirements and where you stand now:

The design must include:
- Event name  ✅
- In-page navigation  -- this [Glitch project](https://glitch.com/edit/#!/w12-demo) shows how to handle the navigation 👍🏻
- Event schedule  
- Details for at least three talks (including talk name and description)  ✅
- Three speaker bios (including a photo)  ✅ 
- Call to action and link to buy tickets  

Your code must include the following HTML elements:
- `header`  
- `main`  
- `footer`  
- `h1`  ✅
- `img`  
- `a`

Commit:
🍵💎 Checkpoint
./index.html:22217575/1498

Commit:
🔋📺 Checkpoint
./index.html:22217575/1868

Commit:
👦🌍 Checkpoint
./index.html:22217575/912

Commit:
🛰🐜 Checkpoint
./index.html:22217575/38724

Commit:
🚔🌶 Checkpoint
./index.html:22217575/1299

Commit:
🏝🎤 Checkpoint
./index.html:22217575/8601

Commit:
🎡🍤 Checkpoint
./index.html:22217575/31779

Commit:
🙏🛢 Checkpoint
./index.html:22217575/61459

Commit:
🐸⛽️ Checkpoint
./styles.css:22217575/426

Commit:
👜🎉 Checkpoint
./styles.css:22217575/999

Commit:
🐕🏛 Checkpoint
./styles.css:22217575/93
./index.html:22217575/8

Commit:
🌯☺️ Checkpoint
./styles.css:22217575/605
./index.html:22217575/763

Commit:
🏖🍅 Checkpoint
./styles.css:22217575/124
./index.html:22217575/214

Commit:
🏔🌅 Checkpoint
./styles.css:22217575/1138
./index.html:22217575/39

Commit:
formatting + beginning styles

Commit:
🙉🔨 Checkpoint
./styles.css:22217575/283
./index.html:22217575/66

Commit:
👜🐕 Checkpoint
./index.html:22217575/3
./styles.css:22217575/312

Commit:
⛄️🏨 Checkpoint
./styles.css:22217575/814
./index.html:22217575/321

Commit:
🍶🚲 Checkpoint
./styles.css:22217575/658
./index.html:22217575/297

Commit:
📕🏤 Checkpoint
./styles.css:22217575/102
./index.html:22217575/40

Commit:
🦀👯 Checkpoint
./styles.css:22217575/106
./index.html:22217575/413

Commit:
🍩🌲 Checkpoint
./styles.css:22217575/641
./index.html:22217575/321

Commit:
⛰🏚 Checkpoint
./styles.css:22217575/488

Commit:
🍗⚽️ Checkpoint
./styles.css:22217575/572

Commit:
👟🎍 Checkpoint
./styles.css:22217575/843
./index.html:22217575/18

Commit:
🐒🏪 Checkpoint
./styles.css:22217575/168
./index.html:22217575/231

Commit:
🍶🐇 Checkpoint
./styles.css:22217575/102

Commit:
🌮🚗 Checkpoint
./styles.css:22217575/1379

Commit:
🎠🌛 Checkpoint
./styles.css:22217575/204

Commit:
🎷📇 Checkpoint
./index.html:22217575/939
./styles.css:22217575/34

Commit:
🐄🐄 Checkpoint
./index.html:22217575/19
./styles.css:22217575/18

Commit:
🍀🌲 Checkpoint
./styles.css:22217575/2866

Commit:
🍂⛪️ Checkpoint
./styles.css:22217575/395
./index.html:22217575/172

Commit:
🎋🌖 Checkpoint
./index.html:22217575/38
./styles.css:22217575/391

Commit:
🏬🚂 Checkpoint
./styles.css:22217575/4495
./index.html:22217575/14

Commit:
🗃☄️ Checkpoint
./.glitch-assets:22217575/3433
./index.html:22217575/119
./styles.css:22217575/85

Commit:
🏆⚽️ Checkpoint
./styles.css:22217575/709
./index.html:22217575/712

Commit:
🙍🍏 Checkpoint
./styles.css:22217575/76
./index.html:22217575/245

Commit:
🚂💐 Checkpoint
./index.html:22217575/1214
./styles.css:22217575/86

Commit:
🎙🛎 Checkpoint
./index.html:22217575/160
./styles.css:22217575/424

Commit:
💳👼 Checkpoint
./styles.css:22217575/18

Commit:
🏹🍝 Checkpoint
./styles.css:22217575/68
./index.html:22217575/15

Commit:
🐌🦂 Checkpoint
./styles.css:22217575/168

Commit:
🚋💨 Checkpoint
./styles.css:22217575/308
./index.html:22217575/100

Commit:
🎳🐣 Checkpoint
./styles.css:22217575/5090
./index.html:22217575/347

Commit:
👝💒 Checkpoint
./styles.css:22217575/1016

Commit:
🛤🛩 Checkpoint
./styles.css:22217575/1590

Commit:
🎇🚞 Checkpoint
./styles.css:22217575/383
./index.html:22217575/148

Commit:
🌿📉 Checkpoint
./styles.css:22217575/1013
./index.html:22217575/193

Commit:
👨‍❤️‍👨✌️ Checkpoint
./styles.css:22217575/222

Commit:
🛋🚦 Checkpoint
./styles.css:22217575/476

Commit:
🌟💴 Checkpoint
./styles.css:22217575/318

Commit:
🏨🛫 Checkpoint
./styles.css:22217575/196

Commit:
🏋🎉 Checkpoint
./styles.css:22217575/2157
./index.html:22217575/73

Commit:
👋🐥 Checkpoint
./index.html:22217575/177
./styles.css:22217575/950

Commit:
🚙🍓 Checkpoint
./index.html:22217575/615
./styles.css:22217575/220

Commit:
🎅🌜 Checkpoint
./styles.css:22217575/430

Commit:
😹🖼 Checkpoint
./styles.css:22217575/451
./index.html:22217575/22

Commit:
🎇🌷 Checkpoint
./styles.css:22217575/225
./index.html:22217575/84

Commit:
🎾🔦 Checkpoint
./index.html:22217575/633
./styles.css:22217575/1396

Commit:
🚇🏝 Checkpoint
./styles.css:22217575/199

Commit:
🎟🦄 Checkpoint
./index.html:22217575/272
./styles.css:22217575/1642

Commit:
🐜🌁 Checkpoint
./styles.css:22217575/516

Commit:
⛱🤘 Checkpoint
./styles.css:22217575/1369

Commit:
😚🏌 Checkpoint
./styles.css:22217575/433

Commit:
🎬🔖 Checkpoint
./styles.css:22217575/534

Commit:
🏨🔦 Checkpoint
./styles.css:22217575/9422
./index.html:22217575/83

Commit:
⚾️🚛 Checkpoint
./index.html:22217575/78
./styles.css:22217575/386

Commit:
🏅👩‍👩‍👦 Checkpoint
./styles.css:22217575/154

Commit:
🎻🏙 Checkpoint
./styles.css:22217575/483

Commit:
👮👓 Checkpoint
./styles.css:22217575/2077
./index.html:22217575/71

Commit:
🐝👹 Checkpoint
./styles.css:22217575/130

Commit:
💼📇 Checkpoint
./styles.css:22217575/1078

Commit:
🏉🎸 Checkpoint
./styles.css:22217575/1349
./index.html:22217575/10

Commit:
🌒🍣 Checkpoint
./styles.css:22217575/890

Commit:
🛬🍉 Checkpoint
./styles.css:22217575/3155

Commit:
🍐🌯 Checkpoint
./styles.css:22217575/18029

Commit:
🚖🛬 Checkpoint
./styles.css:22217575/416

Commit:
🍏🎳 Checkpoint
./styles.css:22217575/194
./index.html:22217575/148

Commit:
🏇🖐 Checkpoint
./styles.css:22217575/1390

Commit:
inline-block, top

Commit:
best practices

whwang99 commented:
https://weichenhwang-gd2-project3.glitch.me/#about

I want detailed feedback on my GitHub PR (this option is similar to how I have graded your previous projects)

> Describe something you struggled with during this project and how you overcame it.

I struggled the most with centering text on the dots and the speaker / talk cards. I overcame it by doing outside research and having a really helpful one on one session. 

> Tell me what you're most proud of from project 3.

I'm most proud of the top three dots (title, about, and schedule) as well as the ticket and return to top buttons. I think they're really sweet and cute, but also they are really effective for my design.  

> If you had another week or two to work on this project, what else would you do?

Try to clean up the spacing especially on the dots when moving from mobile to the full site. I'd like to also clean up how the speaker / talks cards oscillate. They feel unstable currently and seem to work only some of the time. 

> Share a new CSS property or technique you used on this project that you did not know or use before.

flex! given, I know a very rudimentary version of it. I'd like to learn it more in depth though. 


## Short(er) answers

It was to create a one page conference page that includes information about a conference we created. It also should have a navigation that moves up and down the page. 

yes, and I sent a commit with the best practice -m.

Commit:
✍️🖇 Checkpoint
./styles.css:22217575/167

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,294 @@
+    </header>
+
+  
+       <a class="buy-tickets-button"  href="#"> Buy Tickets </a>
```
This is a really delightful and eye-catching call to action. Well done! 🏆

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,294 @@
+       <a class="return-to-top" href=#> ^ </a>
+
+    <main>
+      <!--how do you control where the section id starts? beginning of circle-->
```
If by this you mean when you click the nav links and it takes you down to this section, then, yes, it starts at the beginning of this element. You can modify this in CSS by using a property called `scroll-margin`. So, for example, if you added this to your `#schedule` selector:
```css
#schedule {
  scroll-margin: 60px;
}
```

Then, when I click the Schedule link, it would stop 60 pixels _above_ the circle/dot and provide a bit of nice white space in the design.

Note that `scroll-margin` does **not** affect the margins visible on the page -- it's only spacing added when something is navigated to.

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,294 @@
+      <!--how do you control where the section id starts? beginning of circle-->
+      <section id="about">
+      <h2>About</h2>
+      <p class=about-content> for those interested or working with data
```
Your values (so, in this case, the class name) need to be surrounded in double quotes, like so:

```suggestion
      <p class="about-content"> for those interested or working with data
```

Note that this happens a few times in the rest of the HTML, but I'm not going to call it out each time.

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,294 @@
+   
+      <section id="schedule">
+      <h2>Schedule</h2>
+      <div class=schedule-date><p> for: June 13th, 2020</p></div>
```
Since there's only one element inside this `div`, you may be able to go without it by instead adding the class name directly to the `p` element:
```suggestion
      <p class="schedule-date"> for: June 13th, 2020</p>
```

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,294 @@
+      
+      <ul class="speakers-card">
+        <li class="speaker-card">
+          <footer>
```
In my original card demo, the `footer` element had a purpose, but here it's just extra markup. You can remove it here and inside all the other `.speaker-card` items.

angeliquejw commented on index.html:
```diff
@@ -0,0 +1,294 @@
+            <a href="#">A Visual History of Human Knowledge</a>
+          </h2>
+          <footer>
+            <img src="https://cdn.glitch.com/8df19ca1-7577-4634-85b1-40f869d95fff%2F9e003b0a822daba702608136e73f7be001b6b2f2_2880x1620.jpg?v=1608239061491" class="author-photo" alt="manuel lima, in front of a web structure"/>
```
This class name of `author-photo` doesn't make sense in this context since you don't use "author" as a descriptor elsewhere. Maybe `talk-photo` instead?

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  box-sizing: border-box;
+  font-family: "JetBrains Mono", monospace;
+}
+
```
As a way to set some defaults for your whole HTML document, I recommend that every stylesheet include a `body` selector including the following:
- `background` or `background-color`
- `color`
- `font-family`

You can always change these defaults on individual elements or specific classes elsewhere in your stylesheet, but adding them here creates a solid foundation and will make sure your document displays as expected across browsers.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+}
+
+header {
+  border-style: solid;
+  border-width: 5px;
+  border-color: lightblue;
```
Since the border matches the background color, is it achieving anything here? If the `border-width` were more dramatic, I would see it being used in place of `padding` to keep the interior text away from the edges of your circle, but I don't think 5px is doing much work.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  border-style: solid;
+  border-width: 5px;
+  border-color: lightblue;
+  width: 440px;
+  height: 440px;
```
Instead of set measurements here, you might end up with a more adaptive design if you used relative units. Percentages would lead to things not being perfect circles, but you could use `vw` -- viewport width -- to keep the width and height to a 1:1 ratio and `min-width`, `min-height`,`max-width` and `max-height` to control the outer bounds of things so the dots never get too big or too small.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+nav {
+  text-align: center;
+  vertical-align: middle;
+  line-height: 1px;
```
This very small `line-height` value is always a bad sign and can lead to your text overlapping in ways that makes it impossible to read; especially for an element as important as page navigation, you should avoid setting these very tiny line heights. In general, line height should always match or be greater than the px size of the text or, if set in relative sizes, not be less than `0.8` unless the font style requires drastic correction.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  font-size: 12px;
+  color: white;
+}
+a:link {
+  text-decoration: none;
+  color: black;
+}
+a:visited {
+  text-decoration: none;
+  color: black;
+}
```
You can combine these styles, like so:
```suggestion
a, a:visited {
  text-decoration: none;
  color: black;
}
```

The same idea is true for the CSS on lines 43-51.

Also, while you need the `:visited` pseudoclass to override the default visited styles, for styling your links overall, you can leave off the `:link`; this is also true on. lines 43, 53, 70.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  border-width: 2px;
+  text-align: center;
+  position: fixed;
+  top: 755px;
```
If a device is smaller than 755px tall, this link will never appear. Consider setting this using the `bottom` property instead:
```suggestion
  bottom: 15px;
```

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  color: coral;
+  z-index: 10;
+}
+.buy-tickets-button:hover {
```
Adding a `transition` value to your `.buy-tickets-button` selector above (lines 71-85) will make this hover effect smoother.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+h2 {
+  font-size: 24px;
+  line-height: 1;
+  margin: 0px;
```
```suggestion
  margin: 0px;
```
```suggestion
  margin: 0;
```
When writing values of zero, you can [leave off units of measurement](https://mica-web.github.io/learn/css/best-practices#avoid-using-units-on-measurements-of-zero).

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  font-size: 24px;
+  line-height: 1;
+  margin: 0px;
+  padding: 0px;
```
```suggestion
  padding: 0;
```

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  background-color: green;
+  text-align: center;
+  vertical-align: middle;
+  margin: 0px;
```
```suggestion
  margin: 0;
```

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+      padding: 65px;
+}
+
+div.schedule-date {
+  font-weight: bold;
+  color: white;
+}
```
You do not need the `div` in front of your selector here; it doesn't break anything, but it doesn't help, either, and ties this class name to an HTML element that you may want to change later.

```suggestion
.schedule-date {
  font-weight: bold;
  color: white;
}
```

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  font-weight: bold;
+  color: white;
+}
+div.schedule-content {
```
As above.

```suggestion
.schedule-content {
```

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  font-size: 15px;
+  text-align: center;
+  color: white;
+  margin: 0px;
```
```suggestion
  margin: 0;
```

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+}
+.speakers-card {
+  text-align: justify;
+  width: 350px;
```
If the entire Speakers section has a set width (see line 174 above), you shouldn't need to also repeat it here. The child elements should be restricted by the parent's width.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  border-style: solid;
+  border-width: 5px;
+  border-color: black;
+  width: 450px;
```
Here you set the `li` to be larger than the `section` (see line 174) and the `ul` (see line 178). Instead, I would recommend removing the `width` value here and on the `ul` and updating line 174 with this value.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  border-color: black;
+  width: 450px;
+  height: 300px;
+  overflow: scroll;
```
🏆 Love that you figured out how to create this effect.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  padding: 15px;
+}
+#talks {
+  width: 350px;
```
Same note as on the speakers section. You're setting multiple `width` values and overriding this value with the rule you have on line 215 below. It's simpler to set a declared width on the parent element and then let all the elements inside of it just respect that rule.

So, I would update this value to be 450px and delete the `width` rule below on line 215.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  list-style-type: none;
+}
+
+footer.form {
```
As elsewhere, you don't need to combine the HTML element + the class name like this. Just using the class name is fine:
```suggestion
.form {
```

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+.buy-tickets-button:visited {
+  text-decoration: none;
+  color: coral;
+  z-index: 10;
```
You do not need to add the `z-index` on the `:visited` pseudoclass -- that should not be affected by if a link is visited or not; you only need it on line 85 above.
```suggestion
```

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+  border-color: coral;
+  text-align: center;
+  position: fixed;
+  z-index: 10;
```
As above. The only things in your `:hover` pseudoclass should be what changes on hover -- so, if the `border-width` doesn't change, you do not need to add it here.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+
+@media (min-width: 520px) {
+  header {
+    width: 600px;
+    height: 600px;
```
Here, you've set the width and height to be larger than the `min-width` where this rule is applied, which will always force the dots to be larger than the screen at this size. I do not think this is your intended effect.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+    width: 600px;
+    height: 600px;
+    position: relative;
+    left: 100px;
```
This compounds the issue -- because this dot is larger than the screen (600px on a 520px device) _and_ pushed to the left by 100px. Based on the math (600px + 100px), your media query for these styles should be for devices 700px and larger instead.

angeliquejw commented on styles.css:
```diff
@@ -0,0 +1,293 @@
+      align-items: center;
+      justify-content: center;
+  }*/
+}
```
I think there's some basic organization and tidying up that would help your CSS:

1. Make sure all your base elements are styled first, then class names, then media queries.
2. Avoid using HTML + class selectors like `div.class-name` since it only serves to complicate your CSS.
3. When possible, use either shared classes or combined selectors (like `header, #about, #schedule`) to avoid rewriting/duplicating CSS.
4. Make sure that in your pseudoclasses (like `:visited` and `:hover`) and your media queries you're only writing the CSS you need to -- not repeating all styles.

Overall, I think you've attempted a complicated and fun layout and gotten very close to nailing it, which is extra impressive given the limited CSS we covered in this course. 🏆 Great work!

angeliquejw reviewed (approved):
This is great work, Wei - thanks for your creative and fun layout! In addition the code review on GitHub, I also recorded a video of design feedback for you here: https://www.loom.com/share/f87c38a907fc497e9d29a4f660820306

I also mention a remix of your dots in that video, and you can see that work on [Glitch here](https://glitch.com/~wei-remix-dots). Hope this is helpful and gets you diving into more advanced CSS layouts!
